x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/email/email_connector.tsx

```diff
@@ -192,7 +192,7 @@ export const EmailActionConnectorFields: React.FunctionComponent<
               }
             )}
             disabled={readOnly}
-            checked={hasAuth}
+            checked={hasAuth || false}
```

Comment:
`hasAuth` is boolean, so I can't figure out why ` || false` is needed.... 🤔 
If it _can_ be _undefined_ then something seems to be wrong in the typing 😬

Reply:
In the current implementation, the initial alert object has all user input fields as undefined to be able to not trigger validation on the form opening (each input field has a check on undefined to not set a validation error in this case). When onblur event on the field or click save we are triggering invalidating the values - basically setting it to null instead of undefined. 
Do you think we should consider another approach for implementing UI inputs validation triggering?